Fix param converter handling in RESTEasyReactive

geoand opened the pull request:
Fixes: #13780

geoand committed:
Fix param converter handling in RR

Fixes: #13780

gsmet commented:
Please avoid using acronyms that nobody but us will understand in PR titles. Keep in mind that sometimes our users are looking for a particular fix.

Thanks!

geoand commented:
Gotcha :)

geoand committed:
Convert NotFoundExceptionMapper to annotation style

This make it possible to avoid having to retrieve the
header from the Arc Request Context

geoand commented:
@FroMage all tests passing :)

FroMage commented on independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/ServerExceptionMapper.java:
```diff
@@ -45,4 +46,9 @@
+    /**
+     * The priority with which the exception mapper will be executed
+     */
+    int priority() default Priorities.USER;
```
I must be thick, but I don't see where that is wired in by the processor?

geoand replied:
Good point! It had already been wired up: https://github.com/quarkusio/quarkus/blob/master/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveScanningProcessor.java#L321

FroMage replied:
HAHAHA
